_studio/shared/include/mfxvideo++int.h

```diff
@@ -57,7 +58,11 @@ typedef struct {
 
 static inline int operator==(const GUID & guidOne, const GUID & guidOther)
 {
-    return !memcmp(&guidOne, &guidOther, sizeof(GUID));
+    return
+        guidOne.Data1 == guidOther.Data1 &&
+        guidOne.Data2 == guidOther.Data2 &&
+        guidOne.Data3 == guidOther.Data3 &&
+        std::equal(guidOne.Data4, guidOne.Data4 + sizeof(guidOne.Data4), guidOther.Data4);
```

Comment:
This also can be fixed by:
```
#pragma pack(push, 4)
typedef struct {
...
} GUID;
#pragma pack(pop)
```
or by another way to compare Data4. Opinions?

Reply:
I like your variant